ios/ios_jenkins_pipeline_guide.md

```diff
@@ -0,0 +1,300 @@
+# iOS Project Integration Guide
+---
+
+This guide will walk you through getting your iOS project set up on Jenkins.
+
+There is no setup you need to perform on Jenkins itself -- everything is handled via a `Jenkinsfile`, a text file named `Jenkinsfile`, that lives in the root of your repository.
+
+Jenkins is configured to scan Intrepid's Github Organization and look for projects that satisfy the requirements of being considered an iOS project. Those 2 requirements are simply:
+1. Have a repository whose name ends with `-ios`
+1. Have a `Jenkinsfile` in the root of your repo.
+
+---
+# Pipeline Basics
+
+A Pipeline is an abstract concept that describes a set of steps to go from raw code, to a delivered product. In the case of iOS projects, our pipeline handles code review, tests, and deploying binaries to our OTA server -- as well as reports, and metric collection.
+
+It's helpful to understand at a basic level what a Jenkins Pipeline is, but not strictly necessary for getting your project up and running. [Click for more information on Pipelines](https://jenkins.io/doc/book/pipeline/)
+
+Instead of having users write full Jenkinsfiles themselves, the Intrepid Pipeline Library serves as a wrapper, abstracting away all of the complexity of the build script itself, and having users just define their project model.
+
+---
+# Integration Steps
+
+#### 1. Create the Jenkinsfile
+
+The Jenkinsfile is where you define your pipeline. By default, all iOS projects on Jenkins will load the Intrepid Pipeline Library. This gives you immediate access to all the packages and classes within the library if you need to write a non-standard Jenkinsfile. Create the file by adding a text file named `Jenkinsfile` to your repo.
```

Comment:
There may be some confusion between the Library and the Jenkinsfile. 

A Jenkinsfile is just a text file that's created on a per-project basis, and added to the repo by someone on the team. The Jenkinsfile usually has some Java/Groovy code which defines the pipeline (the build code) to be run for the project. There's a bunch of default functions (like `sh` which runs a shell command) available for all Jenkinsfiles.

Our library extends the available methods to use in the Jenkinsfile. In particular, it provides some classes which define the Project model (`XcodeProject`), and a class which acts as a wrapper around all of the Java/Groovy pipeline code you _would_ need to write, if we didn't have a library to wrap it with.

This allows us to greatly simplify the Jenkinsfile that team leads need to write, by only having to define a single model, and then passing it to a function (the wrapper function `xcodePipeline`)